spec/github_helper_spec.rb

```diff
@@ -217,31 +217,77 @@ def get_milestone(milestone_name:)
       allow(Octokit::Client).to receive(:new).and_return(client)
     end
 
-    it 'has the correct dates to code freeze without submission' do
-      comment = "Code freeze: October 22, 2022\nApp Store submission: November 15, 2022\nRelease: October 25, 2022\n"
-      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+    it 'computes the correct dates for standard period' do
+      due_date = '2022-10-20T08:00:00Z'.to_time.utc
+      options = {
+        due_on: '2022-10-20T12:00:00Z',
+        description: "Code freeze: October 20, 2022\nApp Store submission: October 24, 2022\nRelease: October 27, 2022\n"
+      }
 
       expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
-      create_milestone(need_submission: false, milestone_duration: 24, days_code_freeze: 3)
+      create_milestone(due_date: due_date, days_until_submission: 4, days_until_release: 7)
     end
 
-    it 'has the correct dates to code freeze with submission' do
-      comment = "Code freeze: October 22, 2022\nApp Store submission: October 22, 2022\nRelease: October 25, 2022\n"
-      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+    it 'computes the correct dates when the due date is on the verge of a DST day change' do
+      # The PST to PDT (DST) change is made on the second Sunday in March and finishes on the first Sunday in November
+      Time.zone = 'Pacific Time (US & Canada)'
+      due_date = '2022-06-18T23:00:00Z'.to_time
+      options = {
+        due_on: '2022-06-19T12:00:00Z',
+        description: "Code freeze: June 19, 2022\nApp Store submission: June 21, 2022\nRelease: June 22, 2022\n"
+      }
 
       expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
-      create_milestone(need_submission: true, milestone_duration: 19, days_code_freeze: 3)
+      create_milestone(due_date: due_date, days_until_submission: 2, days_until_release: 3)
     end
 
-    def create_milestone(need_submission:, milestone_duration:, days_code_freeze:)
+    it 'computes the correct dates when the due date is on DST but has no day change' do
+      # The PST to PDT (DST) change is made on the second Sunday in March and finishes on the first Sunday in November
+      Time.zone = 'Pacific Time (US & Canada)'
+      due_date = '2022-06-18T22:00:00Z'.to_time
+      options = {
+        due_on: '2022-06-18T12:00:00Z',
+        description: "Code freeze: June 18, 2022\nApp Store submission: June 20, 2022\nRelease: June 21, 2022\n"
+      }
+
+      expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
+      create_milestone(due_date: due_date, days_until_submission: 2, days_until_release: 3)
+    end
+
+    context 'with input validation' do
```

Comment:
Out curiosity, why did you choose to use a `context` here?

Reply:
I thought that will be more organized and improve the readability to have the input validations under the context, instead of having them grouped under the main describe block.  

Although, now that you ask that question, it made me think if this is really useful at all 🤔  
As I do not have any additional behavior to input on those methods, the context here was just misused. 😄

Reply:
I think Gio's question was more about why using `context` vs `describe` to group the tests. Functionally that doesn't make a difference, but semantically, `context` is more for `when [in situation X]` or `with [given use case or situation]`

Reply:
@raafaelima @AliSoftware, a bit of both 😄 

Why grouping the input validation? (You answered that already) And why with `context` vs `describe`?

It is my understanding that `context` is an alias for `describe` but I can't find the code where that happens. The closer I could get to was the DSL [source code](https://github.com/rspec/rspec-core/blob/ea8554afd1a2b63677c6593059fa8f2476181deb/lib/rspec/core/dsl.rb) which is very meta, and the [documentation](https://rubydoc.info/gems/rspec-core/RSpec/Core/DSL) which implies they are.

In particular, the example from the docs use `context` with a "when ..." like Olivier suggested above. 

```ruby
RSpec.describe "something" do
  context "when something is a certain way" do
    it "does something" do
      # example code goes here
    end
  end
end
```

That pattern matches my experience with and mental model of `describe` vs `context` and it's why seeing this code tickled my interest and made me ask the question 😄

Reply:
Some additional doc / reference about `context` vs `describe`:
 - https://rspec.rubystyle.guide/#example-group-naming
 - https://rspec.rubystyle.guide/#context-descriptions
 - https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspeccontextmethod